server/email_verification.go

```diff
@@ -236,6 +328,12 @@ func handleEmailVerifyFunc(verifiedTpl *template.Template, issuer url.URL, keysF
 			}
 			return
 		}
+
+		qr := cbURL.Query()
+		qr.Set("code", code)
+		qr.Set("state", ses.ClientState)
```

Comment:
Correct me if I'm wrong but I think this state won't be valid for the client being redirected to. Since the clients must generate this state string, they should flat out reject ones they haven't created. So dex can't create a valid state.